### PR TITLE
KOGITO-2490: Shortcuts should not be triggered when typing on text inputs

### PR DESCRIPTION
**Pull request by tiagobento**
https://issues.redhat.com/browse/KOGITO-2490

Keyboard Shortcuts are now always bound to `window`. The API offers a way to bind shortcuts to other HTML elements in the DOM.

### CODE REVIEW

File: packages/microeditor-envelope/src/EditorEnvelopeController.tsx [hunk 1]
```diff
@@ -25,33 +25,29 @@ import { SpecialDomElements } from "./SpecialDomElements";
...
+    private readonly renderer: Renderer,
+    private readonly resourceContentEditorCoordinator: ResourceContentEditorCoordinator,
+    private readonly keyboardShortcutsService: DefaultKeyboardShortcutsService
   ) {
     this.renderer = renderer;
     this.editorFactory = editorFactory;
     this.specialDomElements = specialDomElements;
     this.resourceContentEditorCoordinator = resourceContentEditorCoordinator;
     this.stateControl = stateControl;
+    this.keyboardShortcutsService = keyboardShortcutsService;
```
Comment (caponetto) [inline packages/microeditor-envelope/src/EditorEnvelopeController.tsx]: Since the properties get assigned on the parameter list of the constructor, I guess you can safely remove these lines.

File: packages/microeditor-envelope/src/EditorEnvelopeView.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ import * as ReactDOM from "react-dom";
 import * as AppFormer from "@kogito-tooling/core-api";
 import { LoadingScreen } from "./LoadingScreen";
 import { KeyBindingsHelpOverlay } from "./KeyBindingsHelpOverlay";
-import { KeyboardShortcutsApi } from "./api/keyboardShortcuts";
+import {DefaultKeyboardShortcutsService, KeyboardShortcutsApi} from "./api/keyboardShortcuts";
```
Comment (caponetto) [inline packages/microeditor-envelope/src/EditorEnvelopeView.tsx]:
```suggestion
import { DefaultKeyboardShortcutsService, KeyboardShortcutsApi } from "./api/keyboardShortcuts";
```
Just a formatting suggestion. :)

**Review by caponetto — COMMENTED** [review]
Hi @tiagobento! LGTM! I left only a couple of minor suggestions.

File: packages/microeditor-envelope/src/EditorEnvelopeController.tsx (hunk 1 above)
Comment (tiagobento) [reply packages/microeditor-envelope/src/EditorEnvelopeController.tsx]: Done in fb26ff2

File: packages/microeditor-envelope/src/EditorEnvelopeView.tsx (hunk 2 above)
Comment (tiagobento) [reply packages/microeditor-envelope/src/EditorEnvelopeView.tsx]: Done in fb26ff2

**Review by caponetto — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
@tiagobento Hello. I searched unit tests of an `IGNORED_TAGS` logic, however I didn't found it. Maybe it is covered by some existing test?. Could you please point me to unit tests checking the `IGNORED_TAGS` logic? If no such exists please create one.

**Comment by jomarko** [comment]
Just for a record, In the mean time I found a corner case, as the 'input' tag (that we restricted in source code of fix) has no focus once node name popup appears, so for user can still seem as unfixed. I asked for tips in jira https://issues.redhat.com/browse/KOGITO-2490.

**Comment by tiagobento** [comment]
@jomarko Thanks for the review. I just pushed a test for the ignored tags.

**Comment by tiagobento** [comment]
@jomarko About this JIRA, it's an issue I found as well, but I figured it was on Stunner's side. When renaming a node directly on the diagram, the input seems to loose focus when the mouse stops hovering. Glad you opened that issue!

**Comment by jomarko** [comment]
@tiagobento yes, I thought can be solved by this https://github.com/kiegroup/kie-wb-common/pull/3342, however it is not resolving the issue.

**Comment by tiagobento** [comment]
@jomarko Since this issue you found is not related to kogito-tooling, could we move on with this PR and address this issue in a separate one?

**Review by ederign — APPROVED** [review]

**Review by jomarko — APPROVED** [review]
@tiagobento I agree with your proposal. Here is the new issue https://issues.redhat.com/browse/KOGITO-2548.
